File: solutions/go/roman-numerals/1/roman_numerals.go
// Package romannumerals implements a function for roman numerals.
package romannumerals

import "errors"

var numerals = []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}
var values = []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}

// ToRomanNumeral converts a base 10 number into a roman numeral.
func ToRomanNumeral(input int) (result string, _ error) {
	if input < 1 || input > 3999 {
		return "", errors.New("input out of roman numeral range")
	}

	for i := 0; i < len(numerals); i++ {
		for input >= values[i] {
			result += numerals[i]
			input -= values[i]
		}
	}

	return
}

File: solutions/julia/roman-numerals/1/roman-numerals.jl
numerals = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]

function to_roman(number)
    if number <= 0
        throw(ErrorException("error"))
    end
    
    idx = 1
    res = ""

    while number > 0
        while values[idx] <= number
            res *= numerals[idx]
            number = number - values[idx]
        end

        idx = idx + 1
    end

    return res
end
